src/rabbit_mgmt_util.erl

```diff
@@ -110,25 +110,22 @@ user_matches_vhost(ReqData, User) ->
         not_found -> true;
         none      -> true;
         V         ->
-            SocketInfo = authz_socket_info(ReqData),
-            lists:member(V, list_login_vhosts(User, SocketInfo))
+            AuthzData = get_authz_data(ReqData),
+            lists:member(V, list_login_vhosts(User, AuthzData))
     end.
 
 user_matches_vhost_visible(ReqData, User) ->
     case vhost(ReqData) of
         not_found -> true;
         none      -> true;
         V         ->
-            SocketInfo = authz_socket_info(ReqData),
-            lists:member(V, list_visible_vhosts(User, SocketInfo))
+            AuthzData = get_authz_data(ReqData),
+            lists:member(V, list_visible_vhosts(User, AuthzData))
     end.
 
-authz_socket_info(ReqData) ->
-    Host = cowboy_req:host(ReqData),
-    Port = cowboy_req:port(ReqData),
-    Peer = cowboy_req:peer(ReqData),
-    #authz_socket_info{ sockname = {Host, Port},
-                        peername = Peer}.
+get_authz_data(ReqData) ->
+    {PeerAddress, _PeerPort} = cowboy_req:peer(ReqData),
+    {ip, PeerAddress}.
```

Comment:
Shouldn't this be the new map format `#{peeraddr => PeerAddress}` ?

Reply:
Might be worth renaming `Sock` to `AuthzData` in here: https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_access_control.erl#L134

Reply:
It seems https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_direct.erl#L191 and https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_reader.erl#L1232 would need updating too to pass the new map instead.

Reply:
According to https://ninenines.eu/docs/en/cowboy/2.6/manual/cowboy_req.peer/, `cowboy_req:peer/1` can be used both ways.

Reply:
@dcorbacho be sure to be on the `rabbitmq-auth-backend-cache-20` branch in other repos, see [this code](https://github.com/rabbitmq/rabbitmq-server/blob/rabbitmq-auth-backend-cache-20/src/rabbit_access_control.erl#L128-L157) 😄 

I decided to build the authz data map in `check_access_control` and tag the incoming data with `ip` or `socket`

Reply:
Ops, I missed that PR. I'll go over all that again.